doc/nrf/ug_memfault.rst

```diff
@@ -0,0 +1,208 @@
+.. _ug_memfault:
+
+Memfault integration
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+`Memfault`_ is a cloud-based web application with |NCS| compatibility that monitors devices and allows you to debug issues remotely.
+Memfault integration provides the following features to |NCS|:
+
+* `Remote debugging <Memfault debugging_>`_ - Memfault issues can be efficiently resolved with automatic detection, alerts, deduplication, and actionable insights sent through the cloud.
+* `Continuous monitoring <Memfault monitoring_>`_ - Monitors device and fleet-level metrics like connectivity and low power state in real-time dashboards and displays prepopulated metrics for your devices.
+  You can access device level data to resolve bugs faster.
+
+See the `Memfault documentation <Memfault introduction_>`_ for more details.
+
+.. note::
+   Memfault has been integrated in |NCS| since the v1.6.0 release.
+
+Prerequisites
+*************
+
+Before you start the |NCS| integration with Memfault, make sure that the following prerequisites are completed:
+
+   * :ref:`Setup of nRF Connect SDK <getting_started>` environment.
+   * :ref:`Setup of nRF9160 DK <ug_nrf9160_gs>`.
+   * Sign up in the `Memfault registration page`_ and `create a new project in Memfault`_.
+     You will be directed to the Integration guide for Memfault.
+     You receive the ``project key`` here to set in the :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY` Kconfig option.
+     After registration, you can connect up to 100 devices for free.
+
+Solution architecture
+*********************
+
+The SDK provides functionality to collect debug information from your devices in the form of coredumps, error traces, metrics, logs, and more.
+Memfault saves the data to a configurable storage, after which it is broken into chunks and forwarded at intervals or application-specific times to Memfault's cloud solution for further analysis.
+Communication with the Memfault Cloud is handled by APIs available in `Memfault-SDK`_, while the integration code of the SDK in |NCS| provides some additional functionality.
+The Memfault cloud reassembles the data before analyzing and deduplicates the data.
+The overall fleet health of the issues can be examined through the dashboard of the Memfault web application.
+See the `Memfault MCU Guide`_ for more information.
+
+.. figure:: /images/Memfault_architecture.svg
+   :alt: Memfault Solution architecture
+
+   Memfault Solution architecture
+
+.. _using_memfault:
+
+Integration overview
+********************
+
+The SDK version is included in the `west manifest file`_ in |NCS| and is automatically downloaded when running ``west update``.
+By default, it is downloaded to ``nRF/modules/memfault-firmware-sdk/``.
```

Comment:
The current path (`nrf/modules/memfault-firmware-sdk/`) is where the NCS integration resides, ie the code that we in Nordic provide to integrate Memfault. 
The Memfault SDK is by default downloaded by west to `ncs/modules/lib/memfault-firmware-sdk/`